R/create_protocol_helpers.R

```diff
@@ -0,0 +1,595 @@
+  filenames <- paste0(unique(title_numbers), "_", filenames, ".Rmd")
+  # create new chapters
+  file.create(file.path(path_to_protocol, filenames))
+  # and add chapter contents from docx
+  for (chapter in unique(cumsum(is_title))) {
+    chapter_file <- file.path(path_to_protocol, filenames[chapter])
+    chapter_contents <- contents[chapter == cumsum(is_title)]
+    writeLines(
+      text = chapter_contents,
+      con = chapter_file)
+  }
+  # delete the complete Rmd (output of convert_docx_rmd)
+  file.remove(file.path(path_to_protocol, temp_filename))
+  file.remove(file.path(path_to_protocol, temp2_filename))
+}
+
+
+#' Writes a `_bookdown.yml` file
+#'
+#' Creates contents from its arguments and writes to file `_bookdown.yml`
+#'
+#' @param language the language of the book
+#' @param book_filename the filename of the book
+#' @param path_to_protocol the path to the protocol
+#' @param output_dir_rel relative output directory
+#'
+#' @importFrom ymlthis use_yml_file yml_empty yml_bookdown_opts
+#'
+#' @noRd
+write_bookdown_yml <- function(
+    language,
+    book_filename,
+    path_to_protocol,
+    output_dir_rel) {
+  # create a character vector with the names of all rmd_files
+  # in correct order for compilation
+  rmd_files <- c(
+    "index.Rmd",
+    "NEWS.md",
+    list.files(path = path_to_protocol,
+               pattern = "^\\d{2}.+Rmd$"))
+
+
+  if (language == "en") {
+    labels <- list(
+      fig = 'Figure ', # nolint start
+      tab = 'Table ',
+      eq = 'Equation ',
+      thm = 'Theorem ',
+      lem = 'Lemma ',
+      def = 'Definition ',
+      cor = 'Corrolary ',
+      prp = 'Proposition ',
+      ex = 'Example ',
+      proof = 'Proof. ',
+      remark = 'Remark. ')
+  }
+  if (language == "nl") {
+    labels <- list(
+      fig = 'Figuur ',
+      tab = 'Tabel ',
+      eq = 'Vergelijking ',
+      thm = 'Theorema ',
+      lem = 'Lemma ',
+      def = 'Definitie ',
+      cor = 'Bijgevolg ',
+      prp = 'Propositie ',
+      ex = 'Voorbeeld ',
+      proof = 'Bewijs. ',
+      remark = 'Opmerking. ') # nolint end
+  }
+  bookdown_yml <- yml_empty()
+  bookdown_yml <- yml_bookdown_opts(
+    bookdown_yml,
+    book_filename = book_filename,
+    output_dir = output_dir_rel,
+    rmd_files = rmd_files,
+    delete_merged_file = TRUE,
+    language = list(
+      label = labels
+    )
+  )
+
+  use_yml_file(
+    .yml = bookdown_yml,
+    path = file.path(path_to_protocol, "_bookdown.yml"),
+    quiet = TRUE)
+}
+
+
+#' Writes `_output.yml` file
+#'
+#' Creates and writes a `_output.yml` file
+#'
+#' @param language language of the protocol
+#' @param path_to_protocol path to the protocol
+#'
+#' @importFrom ymlthis use_output_yml yml_empty yml_output
+#' @importFrom bookdown gitbook pdf_book
+#' @importFrom xfun read_utf8 write_utf8
+#'
+#' @noRd
+#'
+write_output_yml <- function(language, path_to_protocol) {
+  output_yml <- yml_empty()
+  if (language == "en") {
```

Comment:
the only difference between the languages seems to be the before element. It is trivial to eliminate the if construction.

```
before <- list(en = list(...), nl = list(...))[["language"]
```